services/github/github-labels.service.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { GithubAuthV3Service } = require('./github-auth-service')
+const { documentation, errorMessagesFor } = require('./github-helpers')
+
+const schema = Joi.object({
+  color: Joi.string(),
+}).required()
+
+module.exports = class GithubLabels extends GithubAuthV3Service {
+  static get category() {
+    return 'issue-tracking'
+  }
+
+  static get route() {
+    return {
+      base: 'github/labels',
+      pattern: ':user/:repo/:name',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub labels',
+        namedParams: {
+          user: 'atom',
+          repo: 'atom',
+          name: 'help-wanted',
+        },
+        staticPreview: this.render({ name: 'help-wanted', color: '#159818' }),
+        documentation,
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: ' ',
+    }
+  }
+
+  static render({ name, color }) {
+    return {
+      message: name,
+      color,
+    }
+  }
+
+  async fetch({ user, repo, name }) {
+    return this._requestJson({
+      url: `/repos/${user}/${repo}/labels/${name}`,
+      schema,
+      errorMessages: errorMessagesFor(`${name} (repo or label not found)`),
```

Comment:
I'd prefer to not inject the user provided label name here and just use `repo or label not found` error message. That'd be more consistent with the other GH badges and may also avoid any potential confusion. 

For example if a user has a typo in the GH repo, but has the label name correct:
https://shields-staging-pr-3902.herokuapp.com/github/labels/bagdes/shields/keep-service-tests-green
![](https://shields-staging-pr-3902.herokuapp.com/github/labels/bagdes/shields/keep-service-tests-green)

Reply:
My reasoning for injecting the name was that labels might get renamed/removed while a repository grows and old references might thereby become unresolveable. In this case I thought it might be nice to know which label was referenced, even when the reference becomes broken at some later point.

However, I agree about consistency. Additionally, confusion due to a typo might be a more common problem than confusion due to old references being not valid anymore (I just literally spent 10 minutes trying to understand why your example with /ba**gd**es/shields did not work).

Reply:
Fixed in d30d301472252ef462679ad37e72f1524f2e9b7c

Reply:
> I just literally spent 10 minutes trying to understand why your example with /bagdes/shields did not work

😉